client/head-manager.js

```diff
@@ -30,7 +30,9 @@ export default class HeadManager {
       tags[h.type] = components
     })
 
-    this.updateTitle(tags.title ? tags.title[0] : null)
+    if (tags.title && tags.title[0]) {
```

Comment:
So my main concern here is that this will actually break when you have this situation:

- `_document.js` has a title tag
- `pages/index.js` doesn't set `title`
- `pages/about.js` sets the title using `next/head`

Now you open `/` and see the title set in _document.js. 
Then you click a `next/link` going to `/about`, the title is updated to the title set in `pages/about.js`
Then you click a `next/link` going back to `/`. In this case, the title is not reverted back to the one in `pages _document.js`, but instead keeps the title from `pages/about.js`.

Reply:
The same situation can be outlined without a custom `_document.js` which is even more common.

Reply:
I'm guessing the issue I just outlined issue is the reason adding `title` in _document.js is a bad idea, since then it's only server rendered, which makes no sense since we also do client side routing. The best solution is probably moving the title into `_app.js` or something similar.

Reply:
I have no problem setting a title on a page by page basis. I misunderstood when I saw an empty title tag after placing the title in the document -- I thought it was a bug with Next JS and the title tag rather than improper placement. I think a console warning that titles should be placed in a head tag on the page level would be enough to fix.